internal/docker_test.go

```diff
@@ -0,0 +1,480 @@
+					cntrList = remainingCntr
+
+					return nil
+				},
+
+				ContainerStatsFn: func(containerID string) (types.ContainerStats, error) {
+					return getContainerStats(cntrList, containerID)
+				},
+			},
+		)
+
+		clist, err := dockerClient.GetCurrentContainers()
+		So(err, ShouldBeNil)
+		So(clist, ShouldNotBeNil)
+		So(len(clist), ShouldEqual, 3)
+
+		// Create a docker client with no containers
+		empDockerClient := NewDockerClient(&mockDockerClient{ContainerListFn: func() ([]types.Container, error) {
+			return []types.Container{}, &ContainerError{Type: ErrContainerListEmpty}
+		}})
+
+		emplist, err := empDockerClient.GetCurrentContainers()
+		So(err, ShouldNotBeNil)
+		So(emplist, ShouldBeEmpty)
+		So(len(emplist), ShouldEqual, 0)
+
+		// Create some files containing container id
+		dockerTempDir, err := ioutil.TempDir("/tmp", "docker_temp_")
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		dockerContainerFile := filepath.Join(dockerTempDir, "Container.txt")
+		err = ioutil.WriteFile(dockerContainerFile, []byte("container_id2"), 0600)
+		So(err, ShouldBeNil)
+
+		dockerNewContainerFile := filepath.Join(dockerTempDir, "NewContainer.txt")
+		err = ioutil.WriteFile(dockerNewContainerFile, []byte("container_id4"), 0600)
+		So(err, ShouldBeNil)
+
+		dockerNonExistingFile := filepath.Join(dockerTempDir, "dockerNonExisting.txt")
+
+		dockerWrongCntrFile := filepath.Join(dockerTempDir, "dockerWrongCntr.txt")
+		err = ioutil.WriteFile(dockerWrongCntrFile, []byte("container_id5"), 0600)
+		So(err, ShouldBeNil)
+
+		dockerEmptyFile := filepath.Join(dockerTempDir, "dockerEmpty.txt")
+		err = ioutil.WriteFile(dockerEmptyFile, []byte(""), 0600)
+		So(err, ShouldBeNil)
+
+		Convey("Remember the current container IDs", func() {
+			// For client with empty list of containers
+			err := empDockerClient.RememberCurrentContainerIDs()
+			So(err, ShouldNotBeNil)
+
+			// For client with list of containers
+			err = dockerClient.RememberCurrentContainerIDs()
+			So(err, ShouldBeNil)
+			So(dockerClient.existingContainers["container_id1"], ShouldBeTrue)
+			So(dockerClient.existingContainers["container_id2"], ShouldBeTrue)
+			So(dockerClient.existingContainers["container_id3"], ShouldBeTrue)
+
+			// Check for an unknown container
+			So(dockerClient.existingContainers["container_id4"], ShouldBeFalse)
+
+			// Create and add new dummy container
+			newCntr := types.Container{
+				ID: "container_id4", Names: []string{"/test_container4", "/test_container4_new"},
+			}
+			cntrList = append(cntrList, newCntr)
+
+			clist, err = dockerClient.GetCurrentContainers()
+			So(err, ShouldBeNil)
+			So(clist, ShouldNotBeEmpty)
+			So(len(clist), ShouldEqual, 4)
+
+			So(dockerClient.existingContainers["container_id4"], ShouldBeFalse)
+
+			// Remember the new container and check it gets added to exisiting containers
+			err = dockerClient.RememberCurrentContainerIDs()
+			So(err, ShouldBeNil)
+			So(dockerClient.existingContainers["container_id4"], ShouldBeTrue)
```

Comment:
These tests are repeated below, remove redundancy

Reply:
Updated, and some tests do need to run this multiple times to verify the correct changes.